eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -91,209 +96,183 @@ object Graph {
-    val vertexQueue = new org.jheaps.tree.SimpleFibonacciHeap[WeightedNode, Short](QueueComparator)
-
-    // initialize the queue and cost array with the initial weight
-    weight.put(targetNode, initialWeight)
-    vertexQueue.insert(WeightedNode(targetNode, initialWeight))
-
-    var targetFound = false
-
-    while (!vertexQueue.isEmpty && !targetFound) {
-
-      // node with the smallest distance from the source
-      val current = vertexQueue.deleteMin().getKey // O(log(n))
-
-      if (current.key != sourceNode) {
-
-        // build the neighbors with optional extra edges
-        val currentNeighbors = extraEdges.isEmpty match {
-          case true => g.getIncomingEdgesOf(current.key)
-          case false =>
-            val extraNeighbors = extraEdges.filter(_.desc.b == current.key)
-            // the resulting set must have only one element per shortChannelId
-            g.getIncomingEdgesOf(current.key).filterNot(e => extraNeighbors.exists(_.desc.shortChannelId == e.desc.shortChannelId)) ++ extraNeighbors
-        }
-
-        // note: there is always an entry for the current in the 'weight' map
-        val currentWeight = weight.get(current.key)
-
-        // for each neighbor
-        currentNeighbors.foreach { edge =>
-
-          val neighbor = edge.desc.a
-
-          // note: 'newMinimumKnownWeight' contains the smallest known cumulative cost (amount + fees) necessary to reach 'current' so far
-          val newMinimumKnownWeight = edgeWeight(edge, currentWeight, initialWeight.length == 0 && neighbor == sourceNode, currentBlockHeight, wr)
-
-          // test for ignored edges
-          if (edge.update.htlcMaximumMsat.forall(newMinimumKnownWeight.cost <= _) &&
-            newMinimumKnownWeight.cost >= edge.update.htlcMinimumMsat &&
-            boundaries(newMinimumKnownWeight) && // check if this neighbor edge would break off the 'boundaries'
-            !ignoredEdges.contains(edge.desc) && !ignoredVertices.contains(neighbor)
-          ) {
-            // we call containsKey first because "getOrDefault" is not available in JDK7
-            val neighborCost = weight.containsKey(neighbor) match {
-              case false => RichWeight(MilliSatoshi(Long.MaxValue), Int.MaxValue, CltvExpiryDelta(Int.MaxValue), Double.MaxValue)
-              case true => weight.get(neighbor)
-            }
-
-            // if this neighbor has a shorter distance than previously known
-            if (newMinimumKnownWeight.weight < neighborCost.weight) {
-
-              // update the visiting tree
-              prev.put(neighbor, edge)
-
-              // update the queue
-              vertexQueue.insert(WeightedNode(neighbor, newMinimumKnownWeight)) // O(1)
-
-              // update the minimum known distance array
-              weight.put(neighbor, newMinimumKnownWeight)
+  private def dijkstraShortestPath(g: DirectedGraph,
+                                   sender: PublicKey,
```

Comment:
Why do we need this? Dijkstra computes paths from `source -> target` and we shouldn't need anything else.
When we are computing the shortest path then `sender == sourceNode` and we don't need the sender
When computing partial paths then `sender != sourceNode` but we're indeed looking for a path that does **not** go through the sender, hence we never fall into the case of applying 0-fee for the channel because its our own.
WDYT?

Reply:
This is to replace the `isNeighborTarget` and `isPartial` parameters that were previously used. I found them confusing and there was one place where they were incorrectly used...

Reply:
It's true that the dijkstra calls for spur paths may implicitly not need this, but I think it's much simpler to provide it so that it's obvious that `addEdgeWeight` and `pathWeight` do the right thing and exclude the fee for the first hop (and only that first hop).